app/src/main/java/cm/aptoide/pt/home/HomePresenter.java

```diff
@@ -704,9 +708,43 @@ private void handleLoadMoreError() {
                   .getCardId());
         })
         .map(HomeEvent::getBundle)
-        .cast(ActionBundle.class)
-        .flatMapCompletable(bundle -> home.setupAppComingSoonNotification(bundle.getActionItem()
-            .getUrl()))
+        .cast(AppComingSoonPromotionalBundle.class)
+        .flatMap(bundle -> home.setupAppComingSoonNotification(bundle.getActionItem()
+            .getUrl())
+            .andThen(Observable.just(bundle)))
+        .doOnNext(bundle -> {
+          bundle.setRegisteredForNotification(true);
```

Comment:
I don't agree with this manipulation here. There are 3 entities currently manipulating this field:
- The repo/service which creates the object with the default (false)
- This presenter which is changing it for the notify click and cancel notify click.
- The adapter to update the bundle in the view.

We could at least just pass the value down to the adapter and cut the 2nd manipulation which seems a bit unnecessary in my opinion.